lib/RefBackend/JITHelpers/JITModule.cpp

```diff
@@ -73,19 +75,70 @@ static refbackrt::MutableArrayRef<T> toRefbackrt(llvm::MutableArrayRef<T> a) {
   return refbackrt::MutableArrayRef<T>(a.data(), a.size());
 }
 
+static std::string stringifyShape(refbackrt::ArrayRef<std::int32_t> extents) {
+  static constexpr char *kDynamicDimAsString = "?";
+  std::stringstream ss;
+  ss << "(";
+  for (int i = 0; i < extents.size(); i++) {
+    if (extents[i] < 0)
+      ss << kDynamicDimAsString;
+    else
+      ss << extents[i];
+    if (i != extents.size() - 1)
+      ss << "x";
+  }
+  ss << ")";
+  return ss.str();
+}
+
```

Comment:
@silvasean lmk if using `std::string` and `std::stringstream` here is too heavy handed. It looks like `JITModule.cpp` already pulls in a bunch of LLVM dependencies (llvm::Expected, llvm::Twine, etc.), and I don't think I can do this sort of string operation with llvm::Twine (since there is no operator += for Twine).

Reply:
JITModule.cpp can use "anything" (LLVM or std dependencies). So this is fine.